File: ros_ws/code/turtle_pose_subscriber.py
#!/usr/bin/env python
#use the default python compiler

# import ROS python library
import rospy

#import the Pose message type
from turtlesim.msg import Pose

def pose_callback(msg):
    # print the x and y coordinates, theta angle and linear and angular velocities
    print("x = {}, y = {}, theta = {}".format(msg.x, msg.y, msg.theta))
    print("linear velocity = {}, angular velocity = {}".format(msg.linear_velocity, msg.angular_velocity))
    
# initialize a node named 'pose_subscriber'
rospy.init_node("pose_subscriber")

# creates a subscriber that listens for Pose messages from the topic '/turtle1/pose'
# when a Pose message is recieved, 'pose_callback' function is called
rospy.Subscriber('/turtle1/pose',Pose,pose_callback)

# keep the node executing infinitely
rospy.spin()  

File: ros_ws/code/turtle_pose_subscriber.cpp
// Include the necessary ROS header file for C++ programming
#include <ros/ros.h>

// Include the header file for the Pose message type
#include <turtlesim/Pose.h>

void pose_callback(const turtlesim::Pose::ConstPtr & msg){
    // print the x and y coordinates, theta angle and linear and angular velocities
    std::cout << "x = " << msg->x << ", y = " << msg->y 
	          << "theta = " << msg->theta << std::endl;
    std::cout << "linear velocity = " << msg->linear_velocity 
	          << ", angular velocity = " << msg->angular_velocity << std::endl;
}

int main(int argc, char** argv){
	//initialize a node named 'pose_subscriber'
	ros::init(argc,argv,"pose_subscriber");
	
	//create a nodeHandle object to interface with the ROS network.
	ros::NodeHandle nh;

	// creates a subscriber that listens for Pose messages from the topic 
	// '/turtle1/pose' with a queue size of 10 
	// when a Pose message is recieved, 'pose_callback' function is called
	ros::Subscriber sub= nh.subscribe("/turtle1/pose",10,pose_callback);

	// keep the node executing infinitely
	ros::spin();
	
	return 0;
}

